### ISSUE TITLE
[KEYBOARD]: Community Care service type typeahead loses focus after 1-2 seconds

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#487]
## Issue
The Community Care service type typeahead loses focus after one to two seconds when users first `TAB` into the field. This was noted as an SC 3.2.1 violation. Animated GIF showing steps to recreate attached below.

## Audit Finding
* Note 4, Defect 4

## Acceptance Criteria
* As a keyboard user, I want the Service Type typeahead to retain focus until I have picked a list item. I should be able to tab into the field, pause, and start typing without focus changing.

## Environment
* MacOS Mojave
* Chrome latest
* https://staging.va.gov/find-locations/

## WCAG or Vendor Guidance (optional)
* [On Focus: Understanding SC 3.2.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-receive-focus.html)

## Screenshots or Trace Logs
![focus-change.gif](https://images.zenhubusercontent.com/5ac217b74b5806bc2bcd3fc8/cb682cc4-21ca-4244-ac64-c6d23a56d527)

**Comment by Karak888** [comment]
@1Copenut this is an interesting one as Community Care dropdown should not have Service Type = *Required.  We are getting ready to do some work on CC FL, with the DB specifically, and it is a known defect.  It is interesting though that when you first select Community Cares,  it does not say required selection is made in dropdown...

**Comment by aeltarhoni** [comment]
@Karak888

**Comment by 1Copenut** [comment]
@Karak888 && @aeltarhoni I think the issue captured here might have to do with the [Downshift](https://github.com/downshift-js/downshift) implementation for the Community Care typeahead. It has its own set of internal actions that might be crossing paths with some of our custom code. That's a guess, but I feel a decent one since the third option is a `<select>` menu or disabled unless users search for Community Care from the second dropdown.

**Comment by Karak888** [comment]
@1Copenut just want to confirm who neeeds to work on this and is it ready to move into backlog or do we need more investigation, and if so that would be a FE engineer correct?  Thanks!

**Comment by 1Copenut** [comment]
@Karak888 I think this one's ready to move into Backlog. It'll be an FE ticket. Some event is firing that bumps focus to the top of the page.

**Comment by brangi** [comment]
Cannot reproduce

**Comment by jenstrickland** [comment]
@brangi Here's a screen recording of the bug.
[facility-locator-typeahead-bug.pptx](https://github.com/department-of-veterans-affairs/va.gov-team/files/3848224/facility-locator-typeahead-bug.pptx)

**Comment by brangi** [comment]
@jenstrickland thanks, I can see the issue now

**Comment by jenstrickland** [comment]
@mmiddaugh I just tested this, and the behavior is still happening.

**Comment by mmiddaugh** [comment]
@brangi please confirm with @jenstrickland when this has been merged so she can retest. Thank you!

**Comment by brangi** [comment]
@jenstrickland  @mmiddaugh, fix has been deployed.

**Comment by jenstrickland** [comment]
@brangi The 'combobox' no longer loses focus. However, @mmiddaugh, there is a new issue. The select element is insufficient for showing the options. The options are too long to appear within the select element, and it is impossible to discern what the options are. If it is possible to use `<optgroup>` to break items into groups so that the options are never longer than 23 characters would prevent this issue. On mobile, a select element can only display about 23 characters. This may need to be a new issue, or an item to solve in the design system. The select element is not usable in this implementation at present.

<img width="1103" alt="facility-locator-dropdown-issue" src="https://user-images.githubusercontent.com/57469/70461282-527bb400-1a86-11ea-91c0-33ab6c5e18ab.png">

<img width="1057" alt="facility-locator-dropdown-issue-selected" src="https://user-images.githubusercontent.com/57469/70461300-5ad3ef00-1a86-11ea-9cec-9c050a98f111.png">

**Comment by jenstrickland** [comment]
@mmiddaugh - should I create a new issue ticket for this? The select element can't be used the way this is, so trying the `<optgroup>` [more info on the optgroup](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup) would allow this to ship, but I recognize the way the data comes may prevent that implementation.

**Comment by mmiddaugh** [comment]
@jenstrickland, please create a new issue ticket for it. We are working on a redesign that will likely address this one way or another but I don't want to lose your feedback. Thank you!
